src/blueapi/service/interface.py

```diff
@@ -117,26 +120,28 @@ def _update_scan_num(md: dict[str, Any]) -> int:
 
 def setup(config: ApplicationConfig) -> None:
     """Creates and starts a worker with supplied config"""
-
     set_config(config)
-
     set_up_logging(config.logging)
 
     # Eagerly initialize worker and messaging connection
-
     worker()
     stomp_client()
     if numtracker_client() is not None:
         context().run_engine.scan_id_source = _update_scan_num
     _hook_run_engine_and_path_provider()
 
 
-# TODO: https://github.com/DiamondLightSource/blueapi/issues/875
 def _hook_run_engine_and_path_provider() -> None:
-    path_provider = get_path_provider()
+    try:
+        path_provider = get_path_provider()
+    except NameError:
+        path_provider = None
+
     run_engine = context().run_engine
 
-    if isinstance(path_provider, StartDocumentPathProvider):
+    if path_provider is None and numtracker_client() is not None:
```

Comment:
Could: Use `logging.debug` to note if you're not going to configure a path provider

Reply:
shifted this round a bit.